src/AppBundle/Utils/Cart.php

```diff
@@ -51,23 +61,30 @@ class Cart
      */
     private $items = array();
 
+    /**
+     * Cart constructor.
+     * @param Restaurant|null $restaurant
+     */
     public function __construct(Restaurant $restaurant = null)
     {
-        if (null !== $restaurant) {
-            $this->restaurantId = $restaurant->getId();
-        }
-
+        $this->restaurant = $restaurant;
         $this->address = new Address();
     }
 
     public function isForRestaurant(Restaurant $restaurant)
     {
-        return $this->restaurantId === $restaurant->getId();
+        return $this->getRestaurantId() === $restaurant->getId();
```

Comment:
We can't compare instances directly because `$this->restaurant` is unserialized from session whereas `$restaurant` is managed by Doctrine, right?

Reply:
I try to keep the old carts valid - looking into this.